File: client/src/app/core/helpers/paginationHelper.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { map } from 'rxjs/operators';
import { PaginatedResponse } from '../../shared/models/pagination';

export function getPaginatedResponse<T>(url, params, http: HttpClient) {
  var paginatedResponse: PaginatedResponse<T> = new PaginatedResponse<T>();
  return http.get<T>(url, { observe: 'response', params }).pipe(
    map(response => {
      paginatedResponse.items = response.body;
      const pagination = response.headers.get('Pagination')
      if (pagination) {
        paginatedResponse.metaData = JSON.parse(pagination);
      }
      return paginatedResponse;
    })
  );
}

export function getPaginationHeaders(pageNumber: number, pageSize: number) {
  let params = new HttpParams();

  params = params.append('pageNumber', pageNumber.toString());
  params = params.append('pageSize', pageSize.toString());

  return params;
}

File: hybrid/lib/helpers/pagination_helper.dart
import 'dart:convert';
import 'package:dio/dio.dart';
import 'package:hybrid/entities/entities.dart';

class PaginationHelper {
  static Pagination? getPaginatedResponse(Response response) {
    if (response.headers['Pagination'] != null) {
      var paginationJson = response.headers['Pagination']!.join(",");
      Pagination pagination = Pagination.fromJson(jsonDecode(paginationJson));
      return pagination;
    }
    return null;
  }

  static getPaginationHeaders(int pageNumber, int pageSize) {
    Map<String, dynamic> params = {};
    // to append pageNumber and pageSize to params
    return params;
  }
}
